packages/hooks/use-in-view.ts

```diff
@@ -0,0 +1,79 @@
+import { useState, useEffect, useRef, Dispatch, SetStateAction } from "react";
```

Comment:
@DAreRodz could you please take a look at this file? I don't know enough about how `IntersectionObserver` works :(